lib/purescript-cst/src/Language/PureScript/AST/Declarations.hs

```diff
@@ -370,7 +370,7 @@ data Declaration
   -- |
   -- A data type declaration (data or newtype, name, arguments, data constructors)
   --
-  = DataDeclaration SourceAnn DataDeclType (ProperName 'TypeName) [(Text, Maybe SourceType)] [DataConstructorDeclaration]
+  = DataDeclaration SourceAnn DataDeclType (ProperName 'TypeName) [(Text, Maybe SourceType)] [VtaTypeVar] [DataConstructorDeclaration]
```

Comment:
What was the reason for putting these in a separate spine? Did you consider something like `[(Text, Maybe SourceType, VtaTypeVar)]`?

Reply:
Just to reduce breakage for the initial implementation.

Reply:
I'm currently working on shifting the representation to this and I've figured that this encoding is the best approach at the moment. There's too many moving parts in the type-checker code to reliably refactor tuple-based encodings like this, and frankly, it's a bit frustrating to do so /rant.